contracts/src/access_control/simple_write_access_controller.cairo

```diff
@@ -0,0 +1,83 @@
+#[contract]
+mod SimpleWriteAccessController {
+    use starknet::ContractAddress;
+    use starknet::class_hash::ClassHash;
+
+    use chainlink::libraries::access_controller::AccessController;
+    use chainlink::libraries::ownable::Ownable;
+    use chainlink::libraries::upgradeable::Upgradeable;
+
+    #[constructor]
+    fn constructor(owner_address: ContractAddress) {
+        Ownable::initializer(owner_address);
+        AccessController::initializer();
+    }
+
+    #[view]
+    fn has_access(user: ContractAddress, data: Array<felt252>) -> bool {
+        AccessController::has_access(user, data)
```

Comment:
yep i called this out in the pr description - essentially direct off-chain calls always have access. Our current use cases for this function are not impacted, but this shouldn't have any material impact overall

Reply:
In solidity:

```
 * @dev SimpleReadAccessController is not suitable for access controlling writes
 * since it grants any externally owned account access! See
 * SimpleWriteAccessController for that.
```

https://github.com/smartcontractkit/libocr/blob/master/contract2/dev/SimpleReadAccessController.sol#L13-L15

ReadAccessController is only used as a library, and WriteAccessController is used as a contract.

Let's do this:
- Add a `has_read_access()` function that does the `caller.is_zero() || has_access()`
- Use this inside aggregator/aggregator_proxy etc
- Rename `WriteAccessController` to `AccessController` (only expose has_access there)

This way we have a generic access controller that can be deployed, but there's also a specialization of the check in the library for those contracts that need it

It would also cut down on the amount of the code

WDYT?

Reply:
I like that approach 👍 are we also deleting `SimpleReadAccessController` then?

Reply:
Ah yeah, we'd remove `SimpleReadAccessController` 👍🏻